packages/@aws-cdk/aws-stepfunctions-tasks/lib/codebuild/start-build.ts

```diff
@@ -0,0 +1,734 @@
+   */
+  readonly logsConfigOverride?: LogsConfig;
+  /**
+   *  The credentials for access to a private registry.
+   *
+   * @default No override
+   */
+  readonly registryCredentialOverride?: RegistryCredential;
+  /**
+   * The type of credentials AWS CodeBuild uses to pull images in your build.
+   *
+   * @default No override
+   */
+  readonly imagePullCredentialsTypeOverride?: codebuild.ImagePullPrincipalType;
+  /**
+   * Specifies if session debugging is enabled for this build
+   *
+   * @default No override
+   */
+  readonly debugSessionEnabled?: boolean;
+  /**
+   * Enable this flag to override privileged mode in the build project.
+   *
+   * @default No override
+   */
+  readonly privilegedModeOverride?: boolean;
+}
+
+/**
+ * Start a CodeBuild Build as a task
+ *
+ * @see https://docs.aws.amazon.com/step-functions/latest/dg/connect-codebuild.html
+ */
+export class CodeBuildStartBuild extends sfn.TaskStateBase {
+  private static readonly SUPPORTED_INTEGRATION_PATTERNS: sfn.IntegrationPattern[] = [
+    sfn.IntegrationPattern.REQUEST_RESPONSE,
+    sfn.IntegrationPattern.RUN_JOB,
+  ];
+
+  protected readonly taskMetrics?: sfn.TaskMetricsConfig;
+  protected readonly taskPolicies?: iam.PolicyStatement[];
+
+  private readonly integrationPattern: sfn.IntegrationPattern;
+
+  constructor(scope: cdk.Construct, id: string, private readonly props: CodeBuildStartBuildProps) {
+    super(scope, id, props);
+    this.integrationPattern = props.integrationPattern ?? sfn.IntegrationPattern.REQUEST_RESPONSE;
+
+    validatePatternSupported(this.integrationPattern, CodeBuildStartBuild.SUPPORTED_INTEGRATION_PATTERNS);
+
+    this.taskMetrics = {
+      metricPrefixSingular: 'CodeBuildProject',
+      metricPrefixPlural: 'CodeBuildProjects',
+      metricDimensions: {
+        ProjectArn: this.props.project.projectArn,
+      },
+    };
+
+    this.taskPolicies = this.configurePolicyStatements();
+  }
+
+  private configurePolicyStatements(): iam.PolicyStatement[] {
+    let policyStatements = [
+      new iam.PolicyStatement({
+        resources: [this.props.project.projectArn],
+        actions: [
+          'codebuild:StartBuild',
+          'codebuild:StopBuild',
+          'codebuild:BatchGetBuilds',
+          'codebuild:BatchGetReports',
+        ],
+      }),
+    ];
+
+    if (this.integrationPattern === sfn.IntegrationPattern.RUN_JOB) {
+      policyStatements.push(
+        new iam.PolicyStatement({
+          actions: ['events:PutTargets', 'events:PutRule', 'events:DescribeRule'],
+          resources: [
+            cdk.Stack.of(this).formatArn({
+              service: 'events',
+              resource: 'rule/StepFunctionsGetEventForCodeBuildStartBuildRule',
+            }),
+          ],
+        }),
+      );
+    }
+
+    return policyStatements;
+  }
+
+  /**
+   * Provides the CodeBuild StartBuild service integration task configuration
+   */
+  /**
+   * @internal
+   */
+  protected _renderTask(): any {
```

Comment:
the length of this method is somewhat of a signal that we should leverage the `bind` pattern to build up to this  output rather than to do it all inline